src/core/tsi/ssl_transport_security_utils.h

```diff
@@ -142,6 +142,23 @@ tsi_result SslProtectorUnprotect(const unsigned char* protected_frames_bytes,
                                  unsigned char* unprotected_bytes,
                                  size_t* unprotected_bytes_size);
 
+// Verifies the CRL signature using X509_CRL_verify
+// https://www.openssl.org/docs/man3.0/man3/X509_CRL_verify.html
+// return: 1 if valid, 0 if invalid, and -1 is there was a problem
+int verify_crl_signature(X509_CRL* crl, X509* issuer);
+
+// Verifies the CRL issuer and certificate issuer name match
+// return: 0 if equal, otherwise a value per X509_NAME_cmp
+int verify_crl_cert_issuer_names_match(X509_CRL* crl, X509* issuer);
+
+// Verifies the certificate in question has the cRLSign bit present
+// return: true if cRLSign bit is present, 0 otherwise
+bool verify_crl_sign_bit(X509* issuer);
+
+// Gets a stable representation of the issuer name from an X509 certificate
+// return: a std::string of the DER encoding of the X509_NAME issuer name
+std::string IssuerFromCert(X509* cert);
```

Comment:
Rather than returning an empty string on error, could we change the return type to e.g. `std::optional<std::string>` or `absl::StatusOr<std::string>`?

Reply:
Done